### PR TITLE
Update git version tag on local builds.

### PR DESCRIPTION
**Pull request by csyonghe**
We use git tag for release builds on github but not local builds.
This enables using the result of git describe as version string for local builds.

### CODE REVIEW

File: slang-tag-version.h [hunk 1]
```diff
@@ -1 +1 @@
-#define SLANG_TAG_VERSION "unknown"
+#define SLANG_TAG_VERSION "v0.27.6-9-ga05e12fc"
```
Comment (jsmall-nvidia) [inline slang-tag-version.h:1]: Shouldn't this stay as unknown, or be removed from the project?

**Review by jsmall-nvidia — CHANGES_REQUESTED** [review]
The slang-tag-version.h seems to be in .gitignore, but is still part of the project?

**Comment by csyonghe** [comment]
The problem is that slang-tag-version.h is still needed for people who pull and build on linux because we haven't integrated generation of this file as a part of linux build. I am not sure if premake's prebuild event will work there.

**Comment by expipiplus1** [comment]
For other shells, it could be something like this (bananas that the powershell version is so incredibly verbose)

```lua
    filter { "not system:windows" } 
        prebuildcommands {
            'printf "#define SLANG_TAG_VERSION \\"$$(git describe --tags)\\"\\n" > ' .. path.getabsolute("slang-tag-version.h")
        }
```

The prebuild event seems to work fine, although it's extra noise on every build :/

It is however displeasing that the format of this header is duplicated here and in the powershell script. Premake doesn't seem to have any support for template files, but if you could get the powershell equivalent of the (untested) following then I think that'd be neater.

```lua    
        prebuildcommands {
            'sed -e s/@VCS_TAG@/$$(git describe --tags)/' .. 
				        '<' .. path.getabsolute("slang-tag-version.h.in")
				        '>' .. path.getabsolute("slang-tag-version.h")
        }
```

(with slang-tag-version.h.in)
```c
#define SLANG_TAG_VERSION "@VCS_TAG@"
```

**Comment by expipiplus1** [comment]
Also note that, as implemented without a fallback, this makes `git` a build dependency and prevents building from a source tarball.

**Comment by csyonghe** [comment]
I kind of want to use C++17 has_include feature to just allow things to compile without the header to exist. On windows the power shell script is set to not return failure if any of the command fails, so strictly there is not a dependency on git.

**Comment by expipiplus1** [comment]
```lua
    filter { "not system:windows" } 
        prebuildcommands {
            'printf "#define SLANG_TAG_VERSION \\"$$(git describe --tags || printf unknown)\\"\\n" > ' .. path.getabsolute("slang-tag-version.h")
        }
```

This should be tolerant of missing git

**Pull request closed without merging**